src/core_plugins/kibana/public/dashboard/grid.js

```diff
@@ -138,65 +164,45 @@ app.directive('dashboardGrid', function ($compile, Notifier) {
         $scope.$on('globalNav:update', safeLayout);
       }
 
-      // return the panel object for an element.
-      //
-      // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-      // ALWAYS CALL PanelUtils.makeSerializeable AFTER YOU ARE DONE WITH IT
-      // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-      function getPanelFor(el) {
-        const $panel = el.jquery ? el : $(el);
-        const panel = $panel.data('panel');
-        panel.$el = $panel;
-        return panel;
-      }
-
-      // Tell gridster to remove the panel, and cleanup our metadata
-      function removePanel(panel, silent) {
-        // remove from grister 'silently' (don't reorganize after)
-        gridster.remove_widget(panel.$el, silent);
-        panel.$el.removeData('panel');
-        $scope.uiState.removeChild(getPersistedStateId(panel));
-      }
-
       // tell gridster to add the panel, and create additional meatadata like $scope
       function addPanel(panel) {
         PanelUtils.initializeDefaults(panel);
-
         const panelHtml = `
             <li>
                 <dashboard-panel
-                  remove="removePanelFromState(${panel.panelIndex})"
-                  panel="getPanelByPanelIndex(${panel.panelIndex})"
-                  is-full-screen-mode="!chrome.getVisible()"
+                  remove="removePanel(${panel.panelIndex})"
+                  panel="findPanelByPanelIndex(${panel.panelIndex}, panels)"
+                  is-full-screen-mode="isFullScreenMode"
                   is-expanded="false"
-                  get-vis-click-handler="filterBarClickHandler(state)"
-                  get-vis-brush-handler="brushEvent(state)"
-                  save-state="state.save()"
-                  toggle-expand="toggleExpandPanel(${panel.panelIndex})"
-                  create-child-ui-state="createChildUiState(path, uiState)">
+                  get-vis-click-handler="getVisClickHandler(state)"
+                  get-vis-brush-handler="getVisBrushHandler(state)"
+                  save-state="saveState"
+                  toggle-expand="toggleExpand(${panel.panelIndex})"
+                  create-child-ui-state="createChildUiState">
             </li>`;
-        panel.$el = $compile(panelHtml)($scope);
+        const panelElement = $compile(panelHtml)($scope);
+        panelElementMapping[panel.panelIndex] = panelElement;
+        // Put the panelIndex on both the outer jQuery element, and the inner widget.
+        panelElement.panelIndex = panelElement[0].panelIndex = panel.panelIndex;
```

Comment:
I don't think that we should be setting the panelIndex on the jQuery 'wrapper' object. I'm not familiar with the interworking of AngularJS; however, it's possible for a different instance of jQuery/jqLIte to be returned when referring to the same underlying element.

In this context, `panelElement` is an instance of `jQuery.fn.init` while `panelElement[0]` is the actual `HTMLElement` itself.